### DISCUSSION TITLE
Replace `rollup-plugin-dts` with `@rollup/plugin-typescript`

### DISCUSSION THREAD

**Discussion by JayaKrishnaNamburu** [dcastil/tailwind-merge#500] [General]
Currently, i see the types are being rolled up using `rollup-plugin-dts`

https://github.com/dcastil/tailwind-merge/blob/206330efc84f4e50c9608566a358d2925f47742a/scripts/rollup.config.mjs#L74-L89

The issue i am currently facing is, the package merges all the types exports into a single file. As you can see in `dist/types.d.ts` file

```ts
export { type ClassNameValue, type ClassValidator, type Config, type DefaultClassGroupIds, type DefaultThemeGroupIds, type ExperimentalParseClassNameParam, type ExperimentalParsedClassName, createTailwindMerge, extendTailwindMerge, fromTheme, getDefaultConfig, mergeConfigs, twJoin, twMerge, validators_d as validators };
```

The thing is the syntax of `type ClassNameValue` and `type ClassValidator` works only in typescript versions higher than `4.3.0`. So, when using in an older projects. The typescript compiler is breaking because typescript versions lower than `4.3` eg: `4.2.4` seems to bundle the types inside `node_modules` too and it's breaking.

**Comment by dcastil** [comment]
Hey @JayaKrishnaNamburu! 👋

I started using `rollup-plugin-dts` because there were people having issues with imports across `.d.ts` files (ESM / CJS differences), so I need to keep bundling the types into a single file.

I checked if there is an easy way to prevent the inline `type` keywords in the final bundle but that doesn't seem to be the case unfortunately. In case you'll find a way to bundle them differently so that it works in TypeScript <4.3, feel free to submit a PR. Otherwise you could also use [patch-package](https://www.npmjs.com/package/patch-package) in your repo to fix it, but then you'll need to keep the patch up to date with changing tailwind-merge versions.

**Reply by JayaKrishnaNamburu** [reply]
No problem, i can understand. It's too much to support all the possible use-cases lately with `esm` and `cjs`. I ended up upgrading the `typescript`. So, please free to close the discussion.
